src/crawl_client.rs

```diff
@@ -40,27 +41,51 @@ pub async fn run(
 			continue;
 		};
 
-		let positions = block
-			.dimensions
-			.iter_extended_partition_positions(&ENTIRE_BLOCK)
-			.collect::<Vec<_>>();
+		if mode == CrawlMode::Cells || mode == CrawlMode::Both {
```

Comment:
A bit shorter version `if matches!(mode, CrawlMode::Cells | CrawlMode::Both) {`

Reply:
Precisely, 1 character shorter :D

Reply:
But is way better for reading :D

Reply:
And much harder for debugging